tests/test_threadpoolctl.py

```diff
@@ -172,46 +190,49 @@ def test_threadpool_limits_manual_unregister():
         # side-effect.
         limits.unregister()
 
-    assert ThreadpoolController().info() == original_controller.info()
+    assert ThreadpoolController().info() == original_info
 
 
 def test_threadpool_controller_limit():
     # Check that using the limit method of ThreadpoolController only impact its
     # library controllers.
-    original_blas_controller = ThreadpoolController().select(user_api="blas")
-    original_openmp_controller = ThreadpoolController().select(user_api="openmp")
+    blas_controller = ThreadpoolController().select(user_api="blas")
+    original_openmp_info = ThreadpoolController().select(user_api="openmp").info()
 
-    with original_blas_controller.limit(limits=1):
+    with blas_controller.limit(limits=1):
         blas_controller = ThreadpoolController().select(user_api="blas")
-        openmp_controller = ThreadpoolController().select(user_api="openmp")
+        openmp_info = ThreadpoolController().select(user_api="openmp").info()
 
         assert all(
             lib_controller.num_threads == 1
             for lib_controller in blas_controller.lib_controllers
         )
         # original_blas_controller contains only blas libraries so no opemp library
         # should be impacted.
-        assert openmp_controller.info() == original_openmp_controller.info()
+        assert openmp_info == original_openmp_info
 
 
-def test_threadpool_controller_restore():
-    # Check that the restore_limits method of ThreadpoolController is able to set the
-    # limits back to their original values. Similar to
-    # test_threadpool_limits_function_with_side_effect but with the object api
+def test_nested_limits():
+    # Check that exiting the context manager properly restores the original limits even
+    # when nested.
     controller = ThreadpoolController()
+    original_info = controller.info()
```

Comment:
I think we should protect this test to avoid failing in case the host machine has a single CPU. Something in the spirit of the following untested snippet:

```suggestion
    original_info = controller.info()
    if any(info["num_threads"] < 2 for info in original_info):
        pytest.skip("Test requires at least 2 CPUs on host machine")
```

Reply:
I suspect it's unreachable code in our CI, where I think we always have at least 2 cores. But better too safe than not enough